app/questionnaire/router.py

```diff
@@ -258,6 +268,121 @@ def _get_return_to_location_url(
                 "questionnaire.submit_questionnaire", _anchor=return_to_answer_id
             )
 
+    def _get_return_to_for_grand_calculated_summary(
+        self,
+        *,
+        return_to: str | None,
+        return_to_block_id: str | None,
+        location: Location,
+        routing_path: RoutingPath,
+        is_for_previous: bool,
+        return_to_answer_id: str | None = None,
+    ) -> str | None:
+        """
+        Builds the return url for a grand calculated summary,
+        and accounts for it possibly being in a different section to the calculated summaries it references
+        """
+        grand_calculated_summary_section = location.section_id
+        # if return_to_block_id is invalid and no block is found, the section doesn't update but can_access_location will be false, so it still returns None
+        if return_to_block_id and self._schema.get_block(return_to_block_id):
+            # Type ignore: if the block is valid, then we'll be able to find a section for it
+            grand_calculated_summary_section: str = self._schema.get_section_id_for_block_id(return_to_block_id)  # type: ignore
+        if grand_calculated_summary_section != location.section_id:
+            # the grand calculated summary is in a different section which will have a different routing path
+            # BUT - we should not jump back to the grand calculated summary if the current section is not complete
+            # instead go to the next incomplete block in the section
+            if next_incomplete_location := self._get_first_incomplete_location_in_section(
```

Comment:
This is inconsistent between same section vs different section. If diff section, it could return first incomplete, but not if it was same section, then it could return None as location isn't accessible?.

Reply:
Yep, good spot, I've updated the router test to have a test-case for an inaccessible GCS in the same section as well as one in another section, and fixed the issue